internal/ccapi/service_instances.go

```diff
@@ -71,52 +67,39 @@ func (c CCAPI) GetServiceInstances(planGUIDs []string) ([]ServiceInstance, error
 		} `json:"included"`
 	}
 
-	if err := c.requester.Get("v3/service_instances?"+BuildQueryParams(planGUIDs), &receiver); err != nil {
+	if err := c.requester.Get("v3/service_instances?"+BuildQueryParams(getPlansGUIDs(plans)), &receiver); err != nil {
 		return nil, fmt.Errorf("error getting service instances: %s", err)
 	}
 
+	instances := make([]ServiceInstance, 0, len(receiver.Instances))
+
 	// Enrich with service plan, service offering space, and org data
-	servicePlanGUIDLookup := computeServicePlanGUIDLookup(receiver.Included.Plans, receiver.Included.ServiceOfferings)
 	spaceGUIDLookup := computeSpaceGUIDLookup(receiver.Included.Spaces, receiver.Included.Organizations)
-	for i := range receiver.Instances {
-		planName, offeringGUID, offeringName := servicePlanGUIDLookup(receiver.Instances[i].ServicePlanGUID)
-		receiver.Instances[i].ServicePlanName = planName
-		receiver.Instances[i].ServiceOfferingGUID = offeringGUID
-		receiver.Instances[i].ServiceOfferingName = offeringName
-
-		spaceName, orgGUID, orgName := spaceGUIDLookup(receiver.Instances[i].SpaceGUID)
-		receiver.Instances[i].SpaceName = spaceName
-		receiver.Instances[i].OrganizationGUID = orgGUID
-		receiver.Instances[i].OrganizationName = orgName
+	for _, instance := range receiver.Instances {
+		plan := getPlanByGUID(plans, instance.ServicePlanGUID)
+		instance.ServicePlanName = plan.Name
+		instance.ServiceOfferingGUID = plan.ServiceOffering.GUID
+		instance.ServiceOfferingName = plan.ServiceOffering.Name
+		instance.ServicePlanMaintenanceInfoVersion = plan.MaintenanceInfoVersion
+		instance.ServicePlanDeactivated = !plan.Available
+
+		spaceName, orgGUID, orgName := spaceGUIDLookup(instance.SpaceGUID)
+		instance.SpaceName = spaceName
+		instance.OrganizationGUID = orgGUID
+		instance.OrganizationName = orgName
+		instances = append(instances, instance)
```

Comment:
Thanks for this advice. I usually prefer using value semantics instead of pointer semantics, but I think you are right. I know we’re not going to overflow memory by being limited to 5000 instances at most, but I’m going to change it. In this way we will avoid future problems if someone changes the number 5000 to something greater. On the contrary, if we implement pagination, it will be a perfect time to use value semantics